### PR TITLE
FIX : when staticDir doesn't exist, the plugin was processing, output…

### PR DESCRIPTION
**Pull request by jnrdt**
…ing a 404

This throw immediatly an error to the user rather than just printing a file containing a 404 error if the base directory doesn't exist #45

### CODE REVIEW

**Comment by drewlustro** [comment]
@jnrdt - I want to thank you again for this contribution, and I believe I understand what safeguards you made. Could you do me a favor and provide a repro case for the bug referenced in #45 ?

If so, I will move forward with merging this in!

Cheers.

File: index.js [hunk 1]
```diff
@@ -11,42 +11,48 @@ function SimpleHtmlPrecompiler (staticDir, paths, options) {
...
-                html: prerenderedHTML,
-                route: outputPath
-              })
-            }
-            var folder = Path.join(self.options.outputDir || self.staticDir, outputPath)
-            mkdirp(folder, function (error) {
-              if (error) {
-                return reject('Folder could not be created: ' + folder + '\n' + error)
+
+  FS.access(this.staticDir, FS.constants.F_OK, function (err) {
```
Comment (drewlustro) [inline index.js]: ✅ I see the guard against existence of `staticDir` here.

File: index.js [hunk 2]
```diff
@@ -11,42 +11,48 @@ function SimpleHtmlPrecompiler (staticDir, paths, options) {
...
-              FS.writeFile(
-                file,
-                prerenderedHTML,
-                function (error) {
-                  if (error) {
-                    return reject('Could not write file: ' + file + '\n' + error)
-                  }
-                  resolve()
+              var folder = Path.join(self.options.outputDir || self.staticDir, outputPath)
+              mkdirp(folder, function (error) {
```
Comment (drewlustro) [inline index.js]: Looks good, but perhaps use a more descriptive variable name than `folder` ?

**Comment by Tribex** [comment]
Any updates on this? If not I'll close it in a day or two.

**Pull request closed without merging**